pilot/pkg/xds/workload.go

```diff
@@ -77,11 +77,6 @@ func (e WorkloadGenerator) GenerateDeltas(
 		removed = subs.Difference(have).Merge(removed)
 	}
 
-	proxy.Lock()
```

Comment:
It seems weird that we were locking the proxy when modifying/reading `WatchedResource`. Are we sure this lock is safe to remove?

Reply:
We were locking due to the concurrent usage in the /debug interface unfortunately. We have a test covering this. Since we don't modify it now, no lock needed

Reply:
Ah, nice!